### PR TITLE
docs: Document IPsec upgrade issue on v1.13.1

### PR DESCRIPTION
**Pull request by pchaigno**
Cilium upgrades to v1.13.1 can cause severe problems when IPsec is enabled. This commit adds a note with mitigations to the docs.

### CODE REVIEW

File: Documentation/operations/upgrade.rst
Comment (squeed) [inline Documentation/operations/upgrade.rst]: do you mean "staying at v1.13.0?"

File: Documentation/operations/upgrade.rst
Comment (squeed) [inline Documentation/operations/upgrade.rst:319]: when should this be run?

Comment (pchaigno) [reply Documentation/operations/upgrade.rst:319]: When the issue is hit.

File: Documentation/operations/upgrade.rst
Comment (joestringer) [inline Documentation/operations/upgrade.rst:313]: What happens when we release 1.13.2? Ideally it would be clear whether v1.13.2 is also affected or not.

Comment (joestringer) [reply Documentation/operations/upgrade.rst:313]: Ah, I see that there is a PR out https://github.com/cilium/cilium/pull/24773 which is presumably expected to block the release. Perhaps we can make it clear after that is merged that the following release is not impacted by the issue.

Comment (pchaigno) [reply Documentation/operations/upgrade.rst:313]: I do intend to update the upgrade guide on each stable branch once the PR is merged. But it's (unsurprisingly?) easier to get reviews on docs PRs than on fix PRs...

**Review by joestringer — APPROVED** [review]

**Pull request merged**
